tests/test_adonis.py

```diff
@@ -281,25 +327,33 @@ def qubit_mapping(self):
             'Charlie': 'QB3'
         }
 
-    def test_arbitrary_names(self, adonis, circuit):
-        with pytest.raises(ValueError):
-            adonis.map_circuit(circuit)
+    def test_routing_circuit_too_large(self, adonis):
+        qubits = cirq.NamedQubit.range(0, 6, prefix='qubit_')
+        circuit = cirq.Circuit([cirq.X(q) for q in qubits])
+        with pytest.raises(ValueError, match='Number of logical qubits is greater than number of physical qubits'):
+            adonis.route_circuit(circuit)
 
-    def test_indexed_names(self, adonis):
-        qubits = cirq.NamedQubit.range(1, 3, prefix='qubit_')
-        circuit = cirq.Circuit(cirq.measure(*qubits))
-        mapped_circuit = adonis.map_circuit(circuit)
+    def test_routing_without_SWAPs(self, adonis, qubits):
+        # circuit graph can be embedded in the Adonis connectivity graph
+        circuit = cirq.Circuit(
+            cirq.CZ(*qubits[0:2]),
+            cirq.CZ(*qubits[1:3]),
+        )
+        new = adonis.route_circuit(circuit)
 
-        assert mapped_circuit.device == adonis
-        assert mapped_circuit.all_qubits() == frozenset(cirq.NamedQubit.range(1, 3, prefix=adonis.QUBIT_NAME_PREFIX))
+        assert len(new.all_qubits()) == 3
+        assert new.all_qubits() <= set(adonis.qubits)
+        #assert len(new) == len(circuit)  # TODO at the moment the routing algo may add unnecessary SWAPs
```

Comment:
```suggestion
```

Reply:
Seems like a feature of the routing algorithm. If we change the algo then we'll update the tests anyway. Or maybe I misunderstood the point of this test case?

Reply:
That's right, it's a feature of the algorithm. The todo just reminds the reader that the particular routing case here is solvable without adding any SWAPs, and the algorithm could perform better.

Reply:
Okay.